### PR TITLE
[Misc] Some minor simplifications to detokenization logic

### PR DESCRIPTION
**Pull request by njhill**
Some simplifications made for clarity while studying the detokenization code (when working on stop string related fixes).

### CODE REVIEW

File: vllm/transformers_utils/detokenizer.py [hunk 1]
```diff
@@ -148,10 +148,7 @@ def decode_sequence_inplace(self, seq: Sequence,
                     )
                     sample_logprob.decoded_token = new_text
 
-        if seq.tokens is None:
```
Comment (njhill) [inline vllm/transformers_utils/detokenizer.py:151]: `seq.tokens` can never be `None` here

**Comment by njhill** [comment]
@Yard1 the other thing I thought would make sense would be to move the `detokenize_incrementally` and `convert_prompt_ids_to_tokens` functions from `tokenizer.py` to `detokenizer.py`. Didn't include that in this PR yet though so as not to obscure the other deltas. WDYT?

**Review by Yard1 — APPROVED** [review]
Looks good, I think we can do the move to detokenizer.py in this PR

**Comment by njhill** [comment]
Thanks @Yard1! Have now pushed another commit moving those functions.

**Comment by njhill** [comment]
@Yard1 good to merge?

**Comment by njhill** [comment]
Going to merge now that I have the power, since this was already approved.

**Pull request merged**